File: BotsSDK/korebot/src/main/java/kore/botssdk/speech/ui/animators/RmsAnimator.java
package kore.botssdk.speech.ui.animators;

import java.util.ArrayList;
import java.util.List;

import kore.botssdk.speech.ui.SpeechBar;

public class RmsAnimator implements BarParamsAnimator {
    final private List<BarRmsAnimator> barAnimators;


    public RmsAnimator(List<SpeechBar> speechBars) {
        this.barAnimators = new ArrayList<>();
        for (SpeechBar bar : speechBars) {
            barAnimators.add(new BarRmsAnimator(bar));
        }
    }

    @Override
    public void start() {
        for (BarRmsAnimator barAnimator : barAnimators) {
            barAnimator.start();
        }
    }

    @Override
    public void stop() {
        for (BarRmsAnimator barAnimator : barAnimators) {
            barAnimator.stop();
        }
    }

    @Override
    public void animate() {
        for (BarRmsAnimator barAnimator : barAnimators) {
            barAnimator.animate();
        }
    }

    public void onRmsChanged(float rmsDB) {
        for (BarRmsAnimator barAnimator : barAnimators) {
            barAnimator.onRmsChanged(rmsDB);
        }
    }
}

File: BotSdk/ui/src/main/java/com/kore/speech/ui/animators/RmsAnimator.kt
package com.kore.speech.ui.animators

import com.kore.speech.ui.SpeechBar

class RmsAnimator(speechBars: List<com.kore.speech.ui.SpeechBar>) : BarParamsAnimator {
    private val barAnimators: MutableList<BarRmsAnimator>

    init {
        barAnimators = ArrayList()
        for (bar in speechBars) {
            barAnimators.add(BarRmsAnimator(bar))
        }
    }

    override fun start() {
        for (barAnimator in barAnimators) {
            barAnimator.start()
        }
    }

    override fun stop() {
        for (barAnimator in barAnimators) {
            barAnimator.stop()
        }
    }

    override fun animate() {
        for (barAnimator in barAnimators) {
            barAnimator.animate()
        }
    }

    fun onRmsChanged(rmsDB: Float) {
        for (barAnimator in barAnimators) {
            barAnimator.onRmsChanged(rmsDB)
        }
    }
}